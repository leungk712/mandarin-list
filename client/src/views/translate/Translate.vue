<template>
  <div>
    <v-container fluid class="translate-container mb-6">
      <v-row class="mt-6">
        <h4>
          Inspired by Google Translate, type in your simplified chinese
          characters and we'll produce a pinyin version
        </h4>
      </v-row>
      <v-row justify="space-around" align="center" class="translate-row mt-6">
        <v-col cols="6">
          <h4>Simplified Chinese Characters | 简体中文</h4>
          <v-textarea
            data-testid="translate-simplified-textarea"
            class="translate-simplified-textarea"
            v-model="characters"
            outlined
            placeholder="你好"
            rows="4"
            no-resize
          />
        </v-col>
        <v-col>
          <h4>Pinyin | 拼音</h4>
          <v-textarea
            data-testid="translate-result-textarea"
            class="translate-result-textarea"
            v-model="converted"
            outlined
            readonly
            placeholder="Nǐ hǎo"
            rows="4"
            no-resize
          />
        </v-col>
      </v-row>
    </v-container>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from "vue-property-decorator";
import pinyin from "chinese-to-pinyin";

@Component({
  name: "Translate"
})
export default class Translate extends Vue {
  // ===== Data ===== //
  public characters = "";

  // ===== Methods ===== //

  // ===== Computed ===== //
  get converted() {
    return pinyin(this.characters);
  }

  // ===== Lifecycle Hooks ===== //
}
</script>

<style scoped>
.translate-container {
  width: 80%;
}

.translate-row {
  border: 1px solid #eee;
  border-radius: 5px;
  width: 100%;
}
</style>
