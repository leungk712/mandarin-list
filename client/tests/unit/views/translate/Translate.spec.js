import { createLocalVue, shallowMount } from "@vue/test-utils";
import Translate from "@/views/translate/Translate.vue";
import Vue from "vue";
import Vuex from "vuex";
import Vuetify from "vuetify";

Vue.use(Vuetify);

const localVue = createLocalVue();
localVue.use(Vuex);

const elem = document.createElement("div");
if (document.body) {
    document.body.appendChild(elem);
};

document.body.setAttribute("data-app", true);

const actions = {};

const mockStore = {};

describe('Translate', () => {
    let wrapper;
    const store = new Vuex.Store({
        modules: mockStore
    });
    const mountFunction = (options, store = mockStore) => {
        return shallowMount(Translate, {
           attachTo: elem,
           localVue,
           store,
           vuetify: new Vuetify(),
           ...options
        });
    };
    describe("template", () => {
        it("should match snapshot", () => {
            wrapper = mountFunction({}, store);
            expect(wrapper.vm.$el).toMatchSnapshot();
            wrapper.destroy();
        });
    });
    describe("script", () => {
        it("converted computed property returns the correct value", async() => {
            wrapper = mountFunction({
                data() {
                    return {
                        characters: "北京"
                    }
                }
            }, store);
            expect(wrapper.vm.converted).toBe("běi jīng");
            wrapper.destroy();
        });
    });
});