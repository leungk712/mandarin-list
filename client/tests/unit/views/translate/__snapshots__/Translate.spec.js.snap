// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Translate template should match snapshot 1`] = `
<div>
  <v-container-stub
    class="translate-container mb-6"
    fluid="true"
    tag="div"
  >
    <v-row-stub
      class="mt-6"
      tag="div"
    >
      <h4>
        
        Inspired by Google Translate, type in your simplified chinese
        characters and we'll produce a pinyin version
      
      </h4>
    </v-row-stub>
     
    <v-row-stub
      align="center"
      class="translate-row mt-6"
      justify="space-around"
      tag="div"
    >
      <v-col-stub
        cols="6"
        tag="div"
      >
        <h4>
          Simplified Chinese Characters | 简体中文
        </h4>
         
        <v-textarea-stub
          backgroundcolor=""
          class="translate-simplified-textarea"
          clearicon="$clear"
          data-testid="translate-simplified-textarea"
          errorcount="1"
          errormessages=""
          loaderheight="2"
          messages=""
          noresize="true"
          outlined="true"
          placeholder="你好..."
          rowheight="24"
          rows="4"
          rules=""
          successmessages=""
          type="text"
          value=""
        />
      </v-col-stub>
       
      <v-col-stub
        cols="6"
        tag="div"
      >
        <h4>
          Pinyin | 拼音
        </h4>
         
        <v-textarea-stub
          backgroundcolor=""
          class="translate-result-textarea"
          clearicon="$clear"
          data-testid="translate-result-textarea"
          errorcount="1"
          errormessages=""
          loaderheight="2"
          messages=""
          noresize="true"
          outlined="true"
          placeholder="Nǐ hǎo..."
          readonly="true"
          rowheight="24"
          rows="4"
          rules=""
          successmessages=""
          type="text"
          value=""
        />
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
</div>
`;
